contrib/go/src/python/pants/contrib/go/tasks/go_compile.py

```diff
@@ -60,9 +73,33 @@ def execute(self):
           lib_binary_map[vt.target] = os.path.join(gopath, 'pkg', self.goos_goarch,
                                                    vt.target.import_path + '.a')
 
-  def _go_install(self, target, gopath):
-    build_flags = re.sub(r'^"|"$', '', self.get_options().build_flags)
-    args = safe_shlex_split(build_flags) + [target.import_path]
+  @classmethod
+  @memoized_method
+  def _get_build_flags(cls, build_flags_from_option, is_flagged, target):
+    """Merge build flags with global < target < command-line order
+
+    Build flags can be defined as globals (in `pants.ini`), as arguments to a Target, and
+    via the command-line.
+    """
+    # If self.get_options().build_flags returns a quoted string, remove the outer quotes,
+    # which happens for flags passed from the command-line.
+    if (build_flags_from_option.startswith('\'') and build_flags_from_option.endswith('\'')) or \
```

Comment:
I ended up changing this to the verbose but explicit case of removing wrapped single and double quotes. Here's the command-line that failed before:

```
./pants binary contrib/go/examples/src/go/server/ \
    --compile-go-build-flags='-v -tags "netgo"'
```

In this case, the quoted arg `"netgo"` had it's right hand double quote removed.